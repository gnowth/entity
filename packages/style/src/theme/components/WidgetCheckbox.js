import { css } from 'styled-components';

import { mixin } from '../../selectors';

// eslint-disable-next-line import/prefer-default-export
export const component_widgetCheckbox_main = css`
  ${mixin({ name: 'disabled' })}
  ${mixin({ name: 'readOnly' })}
`;
