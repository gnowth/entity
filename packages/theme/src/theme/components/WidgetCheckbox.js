// eslint-disable-next-line import/prefer-default-export
export const component_widgetCheckbox_radio = {
  iconComponentPropsFalse: {
    name: 'radio_button_unchecked',
  },
  iconComponentPropsNull: {
    name: 'radio_button_unchecked',
  },
  iconComponentPropsTrue: {
    name: 'radio_button_checked',
  },
  inputType: 'radio',
};
