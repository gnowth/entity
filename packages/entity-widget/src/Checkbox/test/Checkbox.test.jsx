import 'jest-styled-components';
import renderer from 'react-test-renderer';
import React from 'react';

import WidgetCheckbox from '..';

it('renders correctly', () => {
  const tree = renderer.create(
    <WidgetCheckbox />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
