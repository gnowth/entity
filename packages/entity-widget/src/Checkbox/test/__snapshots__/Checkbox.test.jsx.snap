// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1.c1 {
  font-size: 1rem;
}

.c0 {
  display: inline-block;
  line-height: 0;
  position: relative;
}

.c2 {
  bottom: 0;
  cursor: pointer;
  height: 100%;
  left: 0;
  opacity: 0;
  position: absolute;
  right: 0;
  top: 0;
  width: 100%;
}

<div
  className="c0"
>
  <i
    className="material-icons sc-gzVnrw ibZhkv c1"
    name="check_box_outline_blank"
  >
    check_box_outline_blank
  </i>
  <input
    checked={false}
    className="c2"
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;
